server/proxy/proxy.go

```diff
@@ -184,6 +191,16 @@ func (p *Proxy) upstreamForApp(app *apps.App) (upstream.Upstream, error) {
 		return up, nil
 
 	default:
-		return nil, utils.NewInvalidError("not a valid app type: %s", app.AppType)
+		return nil, utils.NewInvalidError("invalid app type: %s", app.AppType)
+	}
+}
+
+func isAppTypeSupported(conf config.Config, m *apps.Manifest) error {
+	if conf.DeveloperMode {
+		return nil
 	}
+	if conf.MattermostCloudMode != (m.AppType == apps.AppTypeAWSLambda) {
+		return utils.NewForbiddenError("AWS lambda apps are supported only in Mattermost Cloud")
+	}
```

Comment:
Similar HTTP apps should only run in On-Prem. Should there be a check to enforce that?